cpp/src/arrow/ipc/options.h

```diff
@@ -67,6 +67,16 @@ struct ARROW_EXPORT IpcWriteOptions {
   /// May only be UNCOMPRESSED, LZ4_FRAME and ZSTD.
   std::shared_ptr<util::Codec> codec;
 
+  /// @brief Minimum space savings percentage required for compression to be applied
+  ///
+  /// Space savings is calculated as (1.0 - compressed_size / uncompressed_size).
+  ///
+  /// For example, if min_space_savings = 0.1, a 100-byte body buffer won't undergo
+  /// compression if its expected compressed size exceeds 90 bytes. If this option isn't
+  /// set, compression will be used indiscriminately. If no codec was supplied, this
+  /// option is ignored.
```

Comment:
Also, can you add that enabling non-zero savings can make the data unreadable for pre-12.0.0 Arrow C++ versions?